Zy/465 neural add neuralnet

zheyxu opened the pull request:
# Pull Request Details

## What issue does this PR address

- NEURAL: Add neuralnet() as an option

- Link to associated issue: #465

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [x] Pre-exisiting lint errors noted: [HERE]
- [x] Integration test `make qtest.tmp` screenshot included in issue
- [x] Tested on device:
  - [ ] Linux
  - [x] MacOS
  - [ ] Windows
- [x] Added two reviewers

## Finalising

Once PR discussion is complete and reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev (gjwgit)

zheyxu committed:
fix error

zheyxu committed:
add choice chip of neural algorithm

zheyxu committed:
neuralnet r

zheyxu committed:
Merge remote-tracking branch 'origin/dev' into zy/465_neural_add_neuralnet

zheyxu committed:
fix

zheyxu committed:
add plot

zheyxu committed:
add integration test

zheyxu committed:
reorder choicechip

zheyxu committed:
Merge remote-tracking branch 'origin/dev' into zy/465_neural_add_neuralnet

zheyxu committed:
remove warning

gjwgit committed:
Update and be consistent with spacing.

gjwgit committed:
Standard title line for model neuralnet R script.

zheyxu committed:
rename tds

zheyxu committed:
Merge remote-tracking branch 'origin/dev' into zy/465_neural_add_neuralnet

zheyxu committed:
rename target

zheyxu committed:
add enable

zheyxu committed:
update parameter name

zheyxu committed:
update hidden layers

zheyxu committed:
add enabled

zheyxu committed:
update label name

zheyxu committed:
add error fct

zheyxu committed:
add act fct

zheyxu committed:
layout redesign

zheyxu committed:
neural threshold

zheyxu committed:
neural step max

zheyxu committed:
add hidden neurons

zheyxu committed:
Merge remote-tracking branch 'origin/dev' into zy/465_neural_add_neuralnet

zheyxu committed:
update tests

zheyxu committed:
add tests

gjwgit committed:
Review and update to required style.

gjwgit reviewed (changes requested):
I have undertaken an initial review.

I have updated various scripts and code to the required level of style.

Please review, understand, and update the rest of the code to work with he new vairable names (providers and TEMPLATE variable).

Thanks.

gjwgit committed:
Merge branch 'dev' into zy/465_neural_add_neuralnet

zheyxu committed:
update var

zheyxu committed:
update nnet

zheyxu committed:
update maxit

zheyxu committed:
update key

zheyxu committed:
update provider

zheyxu committed:
update providers

zheyxu commented:
> I have undertaken an initial review.
> 
> I have updated various scripts and code to the required level of style.
> 
> Please review, understand, and update the rest of the code to work with he new vairable names (providers and TEMPLATE variable).
> 
> Thanks.

Updated.

zheyxu committed:
fix name

gjwgit committed:
Review, update, finalise neuralnet() support

gjwgit committed:
dcm lint fix.
